contrib/go/src/python/pants/contrib/go/tasks/go_protobuf_gen.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import re
+
+from pants.backend.codegen.protobuf.subsystems.protoc import Protoc
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.build_graph.address import Address
+from pants.option.custom_types import target_option
+from pants.task.simple_codegen_task import SimpleCodegenTask
+from pants.util.dirutil import safe_mkdir
+from pants.util.memo import memoized_property
+from pants.util.process_handler import subprocess
+from twitter.common.collections import OrderedSet
+
+from pants.contrib.go.subsystems.protoc_gen_go import ProtocGenGo
+from pants.contrib.go.targets.go_protobuf_library import GoProtobufGenLibrary, GoProtobufLibrary
+
+
+class GoProtobufGen(SimpleCodegenTask):
+
+  _NAMESPACE_PARSER = re.compile(r'^\s*option\s+go_package\s*=\s*"([^\s]+)"\s*;', re.MULTILINE)
+
+  @classmethod
+  def register_options(cls, register):
+    super(GoProtobufGen, cls).register_options(register)
+
+    register('--import-target', type=target_option, advanced=True,
+             help='Target that will be added as a dependency of generated code.')
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(GoProtobufGen, cls).subsystem_dependencies() + (Protoc.scoped(cls), ProtocGenGo.scoped(cls),)
```

Comment:
Since the `Protoc` subsystem is scoped to this class (yay!) I believe you can resolve the version problem by overriding the version of protoc _only_ within this scope... ie: `--protoc-gen-go-proto-version=`.

Likewise, I don't think scoping `ProtocGenGo` makes sense (unless you're expecting reuse in other tasks).

Reply:
Great suggestion about scoped options - I put these in `pants.ini` and we no longer need to pass the protoc version args on the command-line. I bet this will pass CI too.